lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1668,6 +1757,12 @@ def configure_disks(self, vm_obj):
             else:
                 diskspec.device.backing.diskMode = "persistent"
 
+            if 'filename' in expected_disk_spec and expected_disk_spec['filename'] is not None:
+                self.add_existing_vmdk(vm_obj, expected_disk_spec, diskspec, scsi_ctl)
+                continue
```

Comment:
This breaks the unit number ordering.  If you move the unit number increment logic above this block, it fixes the issue.  In addition, I would suggest adding a check for more than 63 disks, since there is a limit of 64 per SCSI controller.

Reply:
This was resolved.